### ISSUE TITLE
[ENH] Improving verbosity on tests on `check_estimator`

### ISSUE THREAD

**Issue by julian-fong** [sktime/sktime#7146]
This issue is being opened to track possible improvements or changes that can be made to help users debug tests when implementing estimators.

As `check_estimator` is the primary tool to verify if estimators are being properly. Improving logging/printing on what tests are being executed, memory management, and others would be helpful for the contributer

**Comment by julian-fong** [comment]
One idea is to switch the `verbose` parameter in `check_estimator` from `True` to an int with various levels `0,1,2,3 ...`. Each level can be specified as to what gets gets logged during the `check_estimator`. The `verbose` parameter would then be forwarded to the `run_tests` function as well. For example:

verbose = `1` could print out the current test that is being ran

**Comment by fkiraly** [comment]
Makes sense to me, and `True` or 1 would be current,`False` or 0 would be none, and 2 would be "more"?
